회원가입 페이지 관련

seongjin2427 opened the pull request:
# Pull Request

- [x] Feature

## Summary

회원가입을 위한 페이지 구현

## Describe your changes

- nickName ,loginId, email, password Input 유효성 검사 적용
- `Join Schema` 추가 작성 
- 별도의 `JoinNotiBar` 컴포넌트 작성 (추후 해당 Notice Bar를 어디에 붙일지 결정 후 삭제 및 변경 예정)

## Issue number and link

- #10 

## ETC

Submit 이벤트로 백엔드 서버로부터 중복체크 후 받는 에러 메시지에 따라 에러 메시지를 띄울 예정입니다.
현재 작성 되어 있는 중복 체크는 임시라고 봐주시면 될 것 같습니다.

seongjin2427 committed:
feat: join form 작성

- 기본적인 회원가입 Form 작성 완료
- nickName, loginId, email, password 유효성 검사 및 중복체크를 위한 로직
  작성
- 알림바를 위한 JoinNotiBar 컴포넌트 작성

* constants/schema.ts
- JOIN_SCHEMA 작성

* constants/message.ts
- SCHEMA_MESSAGES 추가
- DUPLICATE_MESSAGES 추가

seongjin2427 committed:
fix: join page의 styled-components 코드 작성

- 작성했었는데 사라져서 다시 재 입력

seongjin2427 committed:
fix: Schema 변수명 오류로 일부 수정

* constants/message.ts
- SCHEMA_MESSAGES.WRONG_LOGIND_FORMAT
 -> SCHEMA_MESSAGES.WRONG_LOGINID_FORMAT으로 수정

* constants/schema.ts
- JOIN_SCHEMA.loginId의 matches 스키마 메시지 변수 수정
- SCHEMA_MESSAGES.WRONG_LOGIND_FORMAT
 -> SCHEMA_MESSAGES.WRONG_LOGINID_FORMAT

seongjin2427 committed:
fix: 임시 저장

- 변경사항 없음

sangrimlee commented on constants/message.ts:
```diff
@@ -1,3 +1,17 @@
+  WRONG_LOGINID_FORMAT: '아이디에 특수문자 또는 한글이 포함되면 안됩니다!',
+} as const;
+
+type DUPLICATE_MESSAGE_TYPE = {
```
따로 type을 선언해주신 이유가 있을까요?
그리고, type 같은 경우는  Constant랑 헷갈리지 않게 CamelCase로 DuplicateMessageType이라고 하면 더 좋을 것 같습니다.

jma1020 replied:
저도 이 부분에 이렇게 type 꼭 지정해야 돌아갔던건지 궁금합니다!!

sangrimlee commented:
따로 dependencies 수정이 없었는데, yarn.lock이 왜 수정되었나요?
packages-lock.json은 삭제해주시면 될 것 같아요.

seongjin2427 committed:
fix: package-lock.json 삭제, duplicate message type 카멜케이스로 변경

- develop pull conflict 수정
  -> schema.ts - SendEmailSchema type 추가
- package-lock.json 삭제,
- DUPLICATE_MESSAGE_TYPE -> DuplicateMessageType 변수명 변경

seongjin2427 committed:
chore: yarn.lock 원복

- develop의 yarn.lock 가져와서 원복시킴

sangrimlee reviewed (approved):
확인했습니다👍🏼
중복 관련 메시지나 인증 과정에서 필요한 오류 메시지는 나중에 Refactoring 합시다

jma1020 reviewed (approved):
확인했습니다~~~

seongjin2427 committed:
Merge branch 'develop' into feature/join
